ReSharper.FSharp/src/FSharp.Psi/src/Util/SolutionExtension.cs

```diff
@@ -0,0 +1,19 @@
+using JetBrains.Metadata.Reader.API;
+using JetBrains.ProjectModel;
+using JetBrains.ReSharper.Plugins.FSharp.Shim.TypeProviders;
+using JetBrains.ReSharper.Plugins.FSharp.TypeProviders.Protocol.Models;
+
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Util
+{
+  public static class SolutionExtension
+  {
+    public static bool TryGetProvidedType(this ISolution solution, IClrTypeName name,
```

Comment:
Do we really want it as an extension method on a solution? It seems a bit too specific to type providers and not related to a solution in any way. Making it extension on a clr type name would seem a better fit, though maybe a non-extension method would be even better.